app/controllers/album_image_controller.rb

```diff
@@ -0,0 +1,38 @@
+class AlbumImageController < ApplicationController
+  before_action :set_image
+
+  def show
+    http_cache_forever public: true do
+      content_headers_from @image.file.blob
+      stream @image.file.blob
+    end
+  end
+
+  private
+
+  def set_image
+    @image = AlbumImage.friendly.find(params[:id])
+    authorize_image
+  end
+
+  def authorize_image
+    unless current_user.present? && (current_user.site_admin? || logged_in_as_admin_of?(album.circle) || album.shared?)
```

Comment:
I think this line would be complete with only checking for `album.shared?`

Reply:
Otherwise anytime someone/some website would like to reach the image (for example, our blog), there should be a user logged in

Reply:
oh true, i forgot to check for public albums

Reply:
Will correct it tomorrow